### PR TITLE
Add Problems API sample

### LINKED ISSUE #27350: Contribute Problems API project to samples index
We already have a sample, we just need to move it to the samples index.

### PR DESCRIPTION
**Pull request by donat**
Fixes #27350.

Rendered sample page:
image

### CODE REVIEW

File: platforms/documentation/docs/src/docsTest/java/org/gradle/docs/samples/PartitioningSamplesRunner.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ Predicate<String> sampleIdFilter() {
         }
     }
 
-    private static final Predicate<String> SAMPLES_BUCKET = s -> !s.startsWith("snippet-");
+    private static final Predicate<String> SAMPLES_BUCKET = s -> !s.startsWith("snippet-") && s.contains("problems"); // TODO (donat) remove this
```
Comment (hegyibalint) [inline platforms/documentation/docs/src/docsTest/java/org/gradle/docs/samples/PartitioningSamplesRunner.java:71]: :x: Unresolved TODO

File: platforms/documentation/docs/src/samples/templates/problems-api-usage/sample-project/src/main/java/org/gradle/sample/Main.java
Comment (hegyibalint) [inline platforms/documentation/docs/src/samples/templates/problems-api-usage/sample-project/src/main/java/org/gradle/sample/Main.java:1]: ❓Does this file have a reason to exists here?

**Review by hegyibalint — APPROVED** [review]
Excellent work with the Groovy sample!
In many files, I saw missing newlines. I don't know if it's intentional or not - PTAL if you haven't seen them.

File: platforms/documentation/docs/src/docsTest/java/org/gradle/docs/samples/PartitioningSamplesRunner.java (hunk 1 above)
Comment (donat) [reply platforms/documentation/docs/src/docsTest/java/org/gradle/docs/samples/PartitioningSamplesRunner.java]: Fixed

File: platforms/documentation/docs/src/samples/templates/problems-api-usage/sample-project/src/main/java/org/gradle/sample/Main.java
Comment (donat) [reply platforms/documentation/docs/src/samples/templates/problems-api-usage/sample-project/src/main/java/org/gradle/sample/Main.java:1]: No, I'll remove it.

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by hegyibalint** [comment]
Still good to go!

**Comment by donat** [comment]
@bot-gradle merge

**Pull request merged**